de.dlr.sc.virsat.model.extension.tests/src-gen/de/dlr/sc/virsat/model/extension/tests/model/ATestCategoryBase.java

```diff
@@ -144,13 +144,11 @@ private void safeAccessTestReference() {
 		ReferencePropertyInstance propertyInstance = (ReferencePropertyInstance) helper.getPropertyInstance("testReference");
 		CategoryAssignment ca = (CategoryAssignment) propertyInstance.getReference();
 		
-		if (ca != null) {
-			if (testReference == null) {
-				createTestReference(ca);
-			}
-			testReference.setTypeInstance(ca);
-		} else {
+		// Temporary manual fix before all generated code is updated
+		if (ca == null) {
 			testReference = null;
+		} else if (testReference == null || !testReference.getTypeInstance().equals(ca)) {
+			createTestReference(ca);

```

Comment:
We still need to add these changes to the generator here, right?

Reply:
No, because in the new generator this logic is moved to the `BeanPropertyReference`, and the bug doesn't reproduce.